src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int initialCapacity = 16;
+    private final double loadFactor = 0.75;
+    private Node<K, V>[] table = new Node[initialCapacity];
+    private int size;
+    private int capacity = initialCapacity;
 
     @Override
     public void put(K key, V value) {
+        put(key, value,table);
+
+    }
 
+    private void put(K key, V value, Node<K, V>[] table) {
+        Node<K, V> node = new Node<>(key, value, null);
+        int bucketId = getBucketId(key);
+        Node<K, V> nodeTemp = table[bucketId];
+        if (nodeTemp == null) {
+            table[bucketId] = node;
+            size++;
+            return;
+        }
+        while (nodeTemp != null) {
+            if (Objects.equals(nodeTemp.key,key)) {
+                nodeTemp.value = value;
+                return;
+            }
+            if (nodeTemp.next == null) {
+                break;
+            }
+            nodeTemp = nodeTemp.next;
+        }
+        nodeTemp.next = node;
+        if (++size >= capacity * loadFactor) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucket = getBucketId(key);
+        Node<K, V> nodeTemp = table[bucket];
+        while (nodeTemp != null) {
+            if (Objects.equals(nodeTemp.key,key)) {
+                return nodeTemp.value;
+            }
+            nodeTemp = nodeTemp.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+
+        }
+    }
+
+    private void resize() {
+        capacity *= 2;
+        size = 0;
+        Node<K, V>[] nodes = new Node[capacity];
+        for (Node element : table) {
+            Node<K, V> nodeTemp = element;
+            while (nodeTemp != null) {
+                put(nodeTemp.key, nodeTemp.value,nodes);
```

Comment:
in this method you can take another approach: save all values to some `oldBuckets` and operate with that, and assign empty array to `table` (as you do with `nodes` now, and on this line you won't need to pass `nodes`, but `table`. This will eliminate need to have the private method `put`, you will have the regular one with two parameters